frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view_helper.js

```diff
@@ -124,3 +129,67 @@ export async function loadMeshFromFile(
 
   Store.dispatch(finishedLoadingIsosurfaceAction(layerName, id));
 }
+
+type MappingActivationConfirmationProps<R> = {|
+  currentMeshFile: ?APIMeshFile,
+  layerName: ?string,
+  mappingInfo: ActiveMappingInfo,
+  onClick: Function,
+  ...R,
+|};
+
+export function withMappingActivationConfirmation<P, C: ComponentType<P>>(
+  WrappedComponent: C,
+): ComponentType<MappingActivationConfirmationProps<P>> {
+  return function ComponentWithMappingActivationConfirmation(
+    props: MappingActivationConfirmationProps<P>,
+  ) {
+    const { currentMeshFile, layerName, mappingInfo, onClick: originalOnClick, ...rest } = props;
+    const [isConfirmVisible, setConfirmVisible] = useState(false);
+    if (currentMeshFile == null || layerName == null) {
+      return <WrappedComponent {...rest} onClick={originalOnClick} />;
+    }
+
+    const isMappingEnabled = mappingInfo.mappingStatus === MappingStatusEnum.ENABLED;
+    const enabledMappingName = isMappingEnabled ? mappingInfo.mappingName : null;
+    const checkWhetherConfirmIsNeeded = () => {
+      if (enabledMappingName !== currentMeshFile.mappingName) {
+        setConfirmVisible(true);
+      } else {
+        originalOnClick();
+      }
+    };
+
+    const mappingString =
+      currentMeshFile.mappingName != null
+        ? `for the mapping "${
+            currentMeshFile.mappingName
+          }" which is not active. You need to activate the mapping`
+        : "without a mapping but a mapping is active. You need to deactivate the mapping";
+
+    return (
+      <Popconfirm
+        title={`The currently active mesh file "${
+          currentMeshFile.meshFileName
+        }" was computed ${mappingString} before continuing.`}
+        visible={isConfirmVisible}
+        onConfirm={async () => {
+          setConfirmVisible(false);
+          Store.dispatch(setMappingAction(layerName, currentMeshFile.mappingName, "HDF5"));
+          await waitForCondition(
+            () =>
+              getMappingInfo(
+                Store.getState().temporaryConfiguration.activeMappingByLayer,
+                layerName,
+              ).mappingStatus === MappingStatusEnum.ENABLED,
+            100,
+          );
```

Comment:
I'll leave it as-is for now, since resolving the deferred would not be straight-forward. Essentially, the mapping activation is a bit convoluted and consists of multiple actions that are partly triggered by a saga and the deferred would need to be dragged along to be available to be resolved in the very end.

Reply:
Ok, makes sense :+1: